app/src/main/java/org/zephyrsoft/trackworktime/timer/TimeCalculatorV2.java

```diff
@@ -138,17 +144,24 @@ public boolean containsEvents() {
 	private long currentBalance = 0;
 	private int futureWorkDays = -1;
 
-	public TimeCalculatorV2(DAO dao, TimerManager timerManager, LocalDate startDate, boolean handleFlexiTime) {
+	public TimeCalculatorV2(DAO dao, TimerManager timerManager, LocalDate startDate, SharedPreferences preferences) {
 		this.dao = dao;
 		this.timerManager = timerManager;
 
-		this.handleFlexiTime = handleFlexiTime;
+		this.handleFlexiTime = preferences.getBoolean(Key.ENABLE_FLEXI_TIME.getName(), false);
 		if (handleFlexiTime) {
 			this.flexiReset = timerManager.getFlexiReset();
 		} else {
 			this.flexiReset = FlexiReset.NONE;
 		}
 
+		this.roundingEnabled = preferences.getBoolean(Key.ROUNDING_ENABLED.getName(), false);
```

Comment:
`this.` is superfluous here (above it is necessary because the parameter has the same name as the instance variable).